reddit-scraper/src/main/java/de/brandwatch/redditscraper/reddit/RedditScraperForTitles.java

```diff
@@ -0,0 +1,89 @@
+package de.brandwatch.redditscraper.reddit;
+
+import de.brandwatch.redditscraper.kafka.Producer;
+import de.brandwatch.redditscraper.model.Resource;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import java.io.IOException;
+import java.time.ZonedDateTime;
+import java.util.HashSet;
+import java.util.Set;
+
+public class RedditScraperForTitles {
+
+    private static final Logger logger = LoggerFactory.getLogger(RedditScraperForTitles.class);
+
+    private static final String REDDIT_URL = "https://old.reddit.com";
+
+    private Set<String> visitedSites = new HashSet<>();
+
+    private final Producer producer;
+
+    private final RedditScraperForComments redditScraperForComments;
+
+    @Autowired
+    public RedditScraperForTitles(Producer producer, RedditScraperForComments redditScraperForComments) {
+        this.producer = producer;
+        this.redditScraperForComments = redditScraperForComments;
+    }
+
+    @Scheduled(fixedDelay = 1000 * 30)
+    public void scrapeReddit() throws IOException {
+        scrapeRedditForTitles(REDDIT_URL, 1);
+    }
+
+    public void scrapeRedditForTitles(String url, int pageCounter) throws IOException {
+        Document reddit = Jsoup.connect(url).get();
+        Elements posts = reddit.getElementsByClass("top-matter"); //Whitespace at the end is important!
+
+        reddit.getElementsByClass("top-matter").stream()
+                .map(this::mapToResource)
+                .filter(resource -> !(resource.getDate() == null))
+                .forEach(producer::send);
+
+        Elements nextPage = reddit.select(".next-button > a");
+
+        for (Element element : nextPage) {
+            pageCounter = pageCounter + 1;
+            //Scrape top 50 sites
+            if(pageCounter < 51) scrapeRedditForTitles(element.attr("href"), pageCounter);
```

Comment:
This works, but we normally do 

```
if () {

}
```

style. If you don't want to you don't have to change it here, but you might as well get used to it.

Reply:
Changed it